### DISCUSSION TITLE
Verifying JWT token serverside

### DISCUSSION THREAD

**Discussion by wiledal** [appwrite/appwrite#5984] [Q&A]
Hello!

I'm not completely satisfied with the database solution, because of the proprietary ORM, lack of support for advanced queries, and inline permissions.

My idea is to roll a separate API and MongoDB, combining some of the great parts of Appwrite with all the benefits of a proper untethered document store. 

I wish to use JWTs to verify a user's identity, without relying on Appwrite's internal authorization for requests. Ideally I would be able to call something like 

```ts
const jwtValid = await account.validateJWT(jwt)

if (jwtValid) {
  // continue
} else {
  // authorization failed
}
```

Or, do we have to setup and call the API to validate it? as in
```ts
client
    .setEndpoint('xxxx')
    .setProject('xxxx')
    .setJWT(jwt);

const user = await account.get();

if (user) {
  // 🟢
} else {
  // 🔴
}


```

 What's the best approach here?

Thanks!

**Comment by MarcelleBond** [comment]
Hi Wiledal,
If I may ask what approach did you end up taking to validate the JWT token.
